Reconcile migration to v1 fluentd API and conform to latest code.

Cryptophobia opened the pull request:
fix: reconcile upstream with v1 API PR

   - fixing missing code to make v1 API commit work with upstream changes
   - fixing functions for new parameter `force_line_breaks`
   - fixing rubocop errors and exanded lineLength to 100 chars
   - all tests pass on ruby 2.7.4
   - this combines #45 (resolves #45) and reconciles it for the latest code changes in upstream

Signed-off-by: Cryptophobia <aouzounov@vmware.com>

Cryptophobia commented:
All test pass, please merge this @qingling128 !

Cryptophobia commented:
Dear @qingling128 and @igorpeshansky ,

I am beginning to worry about you guys. Is everything okay at Google Cloud?

Yours with PRs ready to merge,
@Cryptophobia

Cryptophobia commented:
I heard about the Google Cloud networking outage yesterday. Is everything okay? 

If the Google Cloud is holding you hostage please blink twice and drop your gpg public keys from the honeypot. We can get you out with the eSWAT team.

lvelvis reviewed (changes requested):
Please note that the current latest code does not support reloading. Therefore, when merging your code, please be mindful of this limitation and ensure that any necessary actions or configurations are taken into consideration to handle this situation appropriately.

nshelke777 reviewed (approved):
Looks good

Cryptophobia commented:
Closes https://github.com/GoogleCloudPlatform/fluent-plugin-detect-exceptions/issues/77

Cryptophobia commented:
@lvelvis @nshelke777 , can you please review and merge in? I have rebased and fixed all rake tests that were failing. 

```
$ rake test
...
Started
Finished in 3.308254697 seconds.
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
28 tests, 2537 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
8.46 tests/s, 766.87 assertions/s
```

Cryptophobia commented:
@qingling128 @igorpeshansky
